solutions/USACO open19-milk-factory.mdx

```diff
@@ -153,3 +155,53 @@ int main(){
 
 </CPPSection>
 </LanguageSection>
+
+## Solution 2
+
+Since there are a total of $n-1$ conveyer belts and a total of $n$ factories, for there to exist a factory that can be traveled to from every other factory, this factory (which we will refer to as the 'special factory') must not have an outgoing conveyer belts. This is because there are a total of $n-i$ other factories that need an outgoing conveyer belt to reach the 'special factory', which leaves no other conveyer belts for the 'special factory' to have, as there are only $n-1$ total conveyer belts. But how do we check if there is a  'special factory'? If there is only one distinct station that appears as an incoming station and not as an outgoing station, then this station is a 'special station'. Otherwise, there is no 'special station', and we should output negative one.
```

Comment:
this is the same method as the official analysis, and since the official solution explains it pretty well, i think you can delete this